scalyr_agent/builtin_monitors/kubernetes_openmetrics_monitor.py

```diff
@@ -0,0 +1,983 @@
+                    log_config=log_config,
+                    scrape_url=kubernetes_api_metrics_scrape_url,
+                )
+
+        # 2. Kubernetes API cAdvisor metrics monitor
+        if self.__scrape_kubernetes_api_cadvisor_metrics:
+            monitor_config, log_config = self.__get_monitor_config_and_log_config(
+                monitor_id=f"{node_name}_kubernetes-api-cadvisor-metrics",
+                url=kubernetes_api_cadvisor_metrics_scrape_url,
+                verify_https=verify_https,
+                ca_file=ca_file,
+                headers=headers,
+                sample_interval=self.__kubernetes_api_cadvisor_metrics_scrape_interval,
+                log_filename=f"openmetrics_monitor-{node_name}-kubernetes-api-cadvisor-metrics.log",
+                metric_name_include_list=self._config.get(
+                    "kubernetes_api_cadvisor_metric_name_include_list"
+                ),
+                metric_name_exclude_list=self._config.get(
+                    "kubernetes_api_cadvisor_metric_name_exclude_list"
+                ),
+                metric_component_value_include_list=self._config.get(
+                    "kubernetes_api_cadvisor_metric_component_value_include_list"
+                ),
+                include_node_name=self.__include_node_name,
+            )
+
+            monitor = monitors_manager.add_monitor(
+                monitor_config=monitor_config,
+                global_config=self._global_config,
+                log_config=log_config,
+            )
+
+            response = monitor.check_connectivity()
+            if response.status_code != 200:
+                self._logger.warn(
+                    f"Kubernetes API cAdvisor metrics endpoint {kubernetes_api_cadvisor_metrics_scrape_url} URL returned non-200 status code {response.status_code}, won't enable this monitor."
+                )
+                monitors_manager.remove_monitor(monitor.uid)
+            else:
+                self.__static_running_monitors.append(monitor.uid)
+                self.__add_watcher_log_config(
+                    monitor=monitor,
+                    log_config=log_config,
+                    scrape_url=kubernetes_api_cadvisor_metrics_scrape_url,
+                )
+
+    def __schedule_dynamic_open_metrics_monitors(self):
+        """
+        Discover a list of metrics exporter URLs to scrape and configure, schedule and run
+        corresponding ScalyrMonitor each scrape url.
+        """
+        start_ts = time.time()
+
+        # 1. Query Kubelet API for pods running on this node and find exporters we want to scrape
+        # (based on the annotations)
+        k8s_pods = self.__get_k8s_pods()
+        node_name = self.__get_node_name()
+
+        self._logger.info(f"Found {len(k8s_pods)} pods on node {node_name}")
+
+        # Maps scrape URL to the corresponding monitor config and K8sPod
+        scrape_configs: Dict[str, Tuple(OpenMetricsMonitorConfig, K8sPod)] = {}
```

Comment:
I guess there have to be square brackets.

Reply:
Yeah, that's type annotations syntax.